### PR TITLE
Custom update 

### PR DESCRIPTION
**Pull request by Daniel-E-B**

### CODE REVIEW

**Comment by aemino** [comment]
what an excellent pr name

File: commands/Cancel.java [hunk 1]
```diff
@@ -1,35 +1,43 @@
 package org.usfirst.frc4904.standard.commands;
```
Comment (NikhilSuresh24) [inline commands/Cancel.java]: don't edit commands

File: custom/CANMessageUnavailableException.java [hunk 2]
```diff
@@ -22,4 +22,5 @@ public CANMessageUnavailableException(String msg, Throwable cause) {
 	public CANMessageUnavailableException(String msg) {
 		super(msg);
 	}
+
```
Comment (NikhilSuresh24) [inline custom/CANMessageUnavailableException.java]: why

File: custom/CommandSendableChooser.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 	 * @param object
 	 */
 	public void addObject(Command object) {
```
Comment (NikhilSuresh24) [inline custom/CommandSendableChooser.java]: take in a commandbase

File: custom/CommandSendableChooser.java [hunk 4]
```diff
@@ -27,6 +27,6 @@ public void addObject(Command object) {
 	 * @param object
 	 */
 	public void addDefault(Command object) {
```
Comment (NikhilSuresh24) [inline custom/CommandSendableChooser.java]: same here, take in a commandbase

File: custom/motioncontrollers/BangBangController.java [hunk 5]
```diff
@@ -65,7 +65,7 @@ public BangBangController(PIDSensor sensor, double A, double F, double threshold
 	 * @param threshold
 	 *        The threshold for the bangbang to start doing something.
 	 */
-	public BangBangController(PIDSource source, double A, double F, double threshold) {
+	public BangBangController(PIDController source, double A, double F, double threshold) {
```
Comment (NikhilSuresh24) [inline custom/motioncontrollers/BangBangController.java]: don't replace `PIDSource` with `PIDController`, just get rid of the constructor

File: custom/motioncontrollers/CustomPIDController.java [hunk 6]
```diff
@@ -91,7 +92,7 @@ public CustomPIDController(double P, double I, double D, double F, PIDSensor sen
 	 * @param sensor
 	 *        The sensor linked to the output
 	 */
-	public CustomPIDController(double P, double I, double D, double F, PIDSource source) {
```
Comment (NikhilSuresh24) [inline custom/motioncontrollers/CustomPIDController.java]: same thing as above with `PIDSource`

File: custom/motioncontrollers/CustomPIDController.java [hunk 7]
```diff
@@ -151,7 +152,7 @@ public CustomPIDController(PIDSensor sensor) {
 	 * @param sensor
 	 *        The sensor linked to the output
 	 */
-	public CustomPIDController(PIDSource source) {
+	public CustomPIDController(PIDController source) {
```
Comment (NikhilSuresh24) [inline custom/motioncontrollers/CustomPIDController.java]: here too

File: custom/motioncontrollers/MotionController.java [hunk 8]
```diff
@@ -81,9 +85,9 @@ public void setOutput(PIDOutput output) {
 	 *        The sensor associated with the output you are
 	 *        trying to control
 	 */
-	public MotionController(PIDSource source) {
-		this(new PIDSensor.PIDSourceWrapper(source));
-	}
+	//public MotionController(PIDController source) {
```
Comment (NikhilSuresh24) [inline custom/motioncontrollers/MotionController.java]: just delete it

File: custom/sensors/CANEncoder.java [hunk 9]
```diff
@@ -27,7 +29,6 @@ public CANEncoder(String name, int id, boolean reverseDirection, double distance
 		this.reverseDirection = reverseDirection;
 		this.distancePerPulse = distancePerPulse;
 		this.offset = 0.0;
-		setPIDSourceType(PIDSourceType.kDisplacement);
```
Comment (NikhilSuresh24) [inline custom/sensors/CANEncoder.java]: setCustomPIDSourceType instead

File: custom/sensors/CANEncoder.java [hunk 10]
```diff
@@ -63,15 +64,8 @@ public CANEncoder(int id) {
...
 	 */
-	@Override
-	public void setPIDSourceType(PIDSourceType pidSource) {
-		this.pidSource = pidSource;
-	}
 
-	@Override
-	public PIDSourceType getPIDSourceType() {
-		return pidSource;
-	}
```
Comment (NikhilSuresh24) [inline custom/sensors/CANEncoder.java]: remake these methods with CustomPIDSourceType

File: custom/sensors/PIDSensor.java [hunk 11]
```diff
@@ -3,7 +3,60 @@
 
 import edu.wpi.first.wpilibj.PIDSource;
 import edu.wpi.first.wpilibj.PIDSourceType;
+import org.usfirst.frc4904.standard.custom.CustomPIDSourceType;
+// import edu.wpi.first.wpilibj2.PIDController
 
+import edu.wpi.first.wpilibj.controller.PIDController;
+
+// public interface PIDSensor {
```
Comment (NikhilSuresh24) [inline custom/sensors/PIDSensor.java]: just delete it

**Comment by bcho04** [comment]
See #260 for the relevant issue for this pull request. This branch fixes #260 once merged.

**Pull request merged**